src/toil/__init__.py

```diff
@@ -145,10 +147,116 @@ def lookupEnvVar(name, envName, defaultValue):
+             will return the appliance string.
+    :rtype: str
+    """
+    appliance = appliance.lower()
+    tag = appliance.split(':')[-1]
+    registry_name = appliance[:-(len(':' + tag))] # remove only the tag
+
+    # jenkins tests
+    if ('quay.io/ucsc_cgl/toil-pr' in registry_name) or ('quay.io/ucsc_cgl/toil-it' in registry_name):
+        return appliance
+
+    # lever to override the check for power users
+    if registry_name.startswith('[override]'):
+        log.warn("Overriding hosted image check.  The image: %s may be unsupported, please "
+                 "be certain it exists or clusters launched may loop forever.  You've been "
+                 "warned." % registry_name + ':' + tag)
+        return registry_name[len('[override]'):] + ':' + tag
+    # docker repo syntax sanity checks
+    elif '://' in registry_name:
+        raise ImageNotFound("Docker images cannot contain a schema (such as '://'): %s"
+                            "" % registry_name + ':' + tag)
+    # docker repo syntax sanity checks
+    elif registry_name[0] == '-':
+        raise ImageNotFound("Docker images cannot begin with '-': %s"
+                            "" % registry_name + ':' + tag)
+    # check quay.io
+    elif 'quay.io/' in registry_name:
+        return requestCheckQuayIo(registry_name=registry_name, tag=tag)
+    # otherwise check docker.io
+    else:
+        return requestCheckDockerIo(registry_name=registry_name, tag=tag)
+
+
+def requestCheckQuayIo(registry_name, tag):
+    """
+    Checks quay.io to see if an image exists using the requests library.
+
+    URL is based on the docker v2 schema.
+
+    :param str registry_name: The url of a docker image's registry.  e.g. "quay.io/ucsc_cgl/toil"
+    :param str tag: The tag used at that docker image's registry.  e.g. "latest"
+    :return: Return True if match found.  Raise otherwise.
+    """
+    # find initial position of 'quay.io/' so that '//', 'www.', etc. are removed
+    clip_position = registry_name.find('quay.io/')
+    # cut off 'quay.io/' itself too to get the path_name
+    path_name = registry_name[clip_position + len('quay.io/'):]
+    if path_name.endswith('/'):
+        path_name = path_name[:-1]
+    quay_url = 'https://quay.io/v2/{path_name}/manifests/{tag}'.format(path_name=path_name, tag=tag)
+    response = requests.head(quay_url)
+    if not response.ok:
+        raise ImageNotFound("TOIL_APPLIANCE_SELF only supports quay.io and official docker.io "
+                            "hosted images.  The image: %s is unsupported (or malformed).  Please "
+                            "supply a docker image with the format: 'quay.io/<repo_path>:<tag>' or "
+                            "'<repo_path>:<tag>'.  Examples: 'quay.io/ucsc_cgl/toil:latest', "
+                            "'ubuntu:latest', or 'broadinstitute/genomes-in-the-cloud:2.0.0'."
+                            "" % registry_name + ':' + tag)
```

Comment:
Should this message include the override option?

Reply:
I'd rather have users make a `quay.io` or `docker.io` account first, rather than try to use `[override]` first, and complain.  They can dig in the docs if they're really bent on some strange docker image, but otherwise they should use the canonical path.  I think it will create fewer complaints for us.  Mostly people should not use `[override]` (or have a reason to).